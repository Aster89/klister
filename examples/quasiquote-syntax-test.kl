#lang "prelude.kl"

(import "quasiquote.kl")

(define thing 'nothing)

(example thing)
(example `thing)
(example `,thing)

(example `(thing))
(example `(,thing))

(example `(list-syntax (,thing thing) thing))

(example `(list-syntax (,thing thing ()) thing))

(example `(thing ,thing thing))
