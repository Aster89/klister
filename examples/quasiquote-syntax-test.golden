#[quasiquote-syntax-test.kl:6.4-8.29]
 <((quasiquote (is quasiquote))
   (unquote (is unquote))
   (unquote-splicing (is unquote-splicing)))> : Syntax
#[quasiquote-syntax-test.kl:11.16-11.23]<nothing> : Syntax
#[quasiquote-syntax-test.kl:14.11-14.16]<thing> : Syntax
#[quasiquote-syntax-test.kl:11.16-11.23]<nothing> : Syntax
#[quasiquote-syntax-test.kl:17.11-17.18]<(thing)> : Syntax
#[quasiquote-syntax-test.kl:18.11-18.19]<(nothing)> : Syntax
#[quasiquote-syntax-test.kl:20.11-20.45]<(list-syntax (nothing thing) thing)> : Syntax
#[quasiquote-syntax-test.kl:22.11-22.48]<(list-syntax (nothing thing ()) thing)> : Syntax
#[quasiquote-syntax-test.kl:24.11-24.31]<(thing nothing thing)> : Syntax
